lib/matplotlib/tests/test_mutate_inputs.py

```diff
@@ -0,0 +1,112 @@
+from matplotlib import pyplot as plt
+import numpy as np
+
+fig = plt.figure()
+ax = fig.add_subplot(111, projection='3d')
+
+x = [1, 2, 3]
+y = [0.0, 0.0, 0.0]
+z = [0.0, 0.0, 0.0]
+
+pl, = ax.plot(x, y, z, 'o-')
+
+ax.set_ylim([0, 4])
+ax.set_zlim([0, 4])
+
+# mutate y,z to get a nontrivial line
+y[:] = [1, 2, 3]
+z[:] = [1, 2, 3]
+
+fig.draw_without_rendering()
+pl.get_data_3d()
+pl.get_data()
+
+
+def test_1():
+
+    result1 = '(array([1, 2, 3]), array([0., 0., 0.]), array([0., 0., 0.]))'
+    result2 = '(array([-0.07348146, -0.02919708,  0.01717111]), ' \
+              'array([-0.04881992, -0.06321353, -0.07828444]))'
+    assert str(pl.get_data_3d()) == result1
+    assert str(pl.get_data()) == result2
```

Comment:
That was the only way i was able to get a matching comparison on the tuple. Is there a better way to do this?

Reply:
You can do something like
```
np.all(p1.get_data() == result2)
```
where result2 is then an np-array.

You may have to do `np.allclose` instead of `np.all` as there may be floating-point effects, at least on `result2`

Reply:
Thanks Oscar. Using `np.allclose` appears to return an assertion error on Ubuntu and Mac:
```
FAILED lib/mpl_toolkits/mplot3d/tests/test_axes3d.py::test_mutating_input_arrays_y_and_z - assert False is True
 +  where False = <function allclose at 0x7f4c5b088ee0>((array([-0.07780068, -0.02919708,  0.02192821]), array([-0.04741606, -0.06321353, -0.07983062])), ((-0.07348146, -0.02919708, 0.01717111), (-0.04881992, -0.06321353, -0.07828444)))
 +    where <function allclose at 0x7f4c5b088ee0> = np.allclose
 +    and   (array([-0.07780068, -0.02919708,  0.02192821]), array([-0.04741606, -0.06321353, -0.07983062])) = <bound method Line2D.get_data of <mpl_toolkits.mplot3d.art3d.Line3D object at 0x7f4c38963730>>()
 +      where <bound method Line2D.get_data of <mpl_toolkits.mplot3d.art3d.Line3D object at 0x7f4c38963730>> = <mpl_toolkits.mplot3d.art3d.Line3D object at 0x7f4c38963730>.get_data
= 1 failed, 8896 passed, 207 skipped, 13 xfailed, 7 xpassed, 5 warnings in 706.06s (0:11:46) =
Error: Process completed with exit code 1.
```
I'm not really sure why this is happening.

Reply:
The values are slightly different:
```
(array([-0.07780068, -0.02919708,  0.02192821]), array([-0.04741606, -0.06321353, -0.07983062]))
```
vs
```
((-0.07348146, -0.02919708, 0.01717111), (-0.04881992, -0.06321353, -0.07828444))
```
I am wondering if setting limits etc will somehow make it different?